File: MCP_GATEWAY/SKill_SOCKET-Mcp_gateway/mcp_server/cerebras_client.py
import os
import httpx  # type: ignore[import-untyped]
class CerebrasClient:
    """LLM client that communicates with the Cerebras API."""

    def __init__(self):
        api_key = os.getenv("CEREBRAS_API_KEY")
        if not api_key:
            raise RuntimeError("CEREBRAS_API_KEY is not set in the environment variables.")
        self.base_url = "https://api.cerebras.ai/v1"
        self.headers = {
            "Authorization": f"Bearer {api_key}",
            "Content-Type": "application/json",
        }

    async def generate_text(self, prompt: str, temperature: float = 0.5) -> str:
        """Send a prompt to the Cerebras chat-completions endpoint and return the text."""
        payload = {
            "model": "llama3.1-8b",
            "messages": [{"role": "user", "content": prompt}],
            "temperature": temperature,
            "max_tokens": 2000,
        }
        try:
            async with httpx.AsyncClient(timeout=30.0) as client:
                response = await client.post(
                    f"{self.base_url}/chat/completions",
                    headers=self.headers,
                    json=payload,
                )
                response.raise_for_status()
                data = response.json()
                return data["choices"][0]["message"]["content"]
        except httpx.HTTPError as exc:
            print(f"Cerebras API Error: {exc}")
            raise RuntimeError("Failed to generate text from Cerebras API.") from exc

File: MCP_GATEWAY/SKill_SOCKET-Mcp_gateway/mcp_server/cerebrasClient.js
const axios = require('axios');

class CerebrasClient {
    constructor() {
        if (!process.env.CEREBRAS_API_KEY) {
            throw new Error("CEREBRAS_API_KEY is not set in the environment variables.");
        }
        this.client = axios.create({
            baseURL: 'https://api.cerebras.ai/v1',
            headers: {
                'Authorization': `Bearer ${process.env.CEREBRAS_API_KEY}`,
                'Content-Type': 'application/json'
            }
        });
    }

    
    async generateText(prompt, temperature = 0.5) {
        try {
            const payload = {
                model: 'llama3.1-8b', // You can change this to other supported models
                messages: [{ role: 'user', content: prompt }],
                temperature,
                max_tokens: 2000,
            };
            const response = await this.client.post('/chat/completions', payload);
            return response.data.choices[0].message.content;
        } catch (error) {
            console.error('Cerebras API Error:', error.response ? error.response.data : error.message);
            throw new Error('Failed to generate text from Cerebras API.');
        }
    }
}

module.exports = CerebrasClient;
